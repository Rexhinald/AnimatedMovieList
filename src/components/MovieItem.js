import React, {useEffect} from "react";
import {motion, useAnimation} from "framer-motion";

const MovieItem = ({movie, finished, flashed, onClick}) => {
  const dateControls = useAnimation();

  const {poster_path, title, release_date, id} = movie;

  const containerVariants = {
    hidden: {
      x: '-100%',
    },
    visible: {
      x: '0%',
      transition: {
        duration: 0.5,
        when: 'beforeChildren',
      }
    }
  };

  const imageVariants = {
    hidden: {
      opacity: 0,
    },
    visible: {
      opacity: 1,
      transition: {
        duration: 0.25,
      }
    }
  };

  const dateVariants = {
    hidden: {
      scale: 1,
    },
    visible: {
      scale: [1.1, 1],
      transition: {
        duration: 1,
      }
    }
  };

  const flashDate = async () => {
    dateControls.start('visible')
  }

  useEffect(() => {
    if(finished) {
      flashDate().then(() => flashed)
    }
  }, [finished])

  return (
    <motion.div
      onClick={onClick}
      variants={containerVariants}
      className="movie-item"
      data-testid={`movie-item-${id}`}
    >
      <motion.img variants={imageVariants} style={{width: 60}} src={`https://image.tmdb.org/t/p/original/${poster_path}`} />
      <div className="movie-item-text">
        <div className="movie-item-title">{title}</div>
        <motion.div variants={dateVariants} initial="hidden" animate={dateControls}>{release_date}</motion.div>
      </div>
    </motion.div>
  );
};

export default MovieItem
