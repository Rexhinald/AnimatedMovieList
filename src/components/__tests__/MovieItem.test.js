import React from "react";
import {cleanup} from "@testing-library/react";
import ReactDOM from "react-dom";
import MovieItem from "../MovieItem";
import renderer from "react-test-renderer";

afterEach(cleanup);

const movie = {
  "adult": false,
  "backdrop_path": "/wETWXzbvn0LhrPGLeRqyIeIl0E8.jpg",
  "genre_ids": [
    18
  ],
  "id": 644089,
  "original_language": "en",
  "original_title": "Blue Bayou",
  "overview": "As a Korean-American man raised in the Louisiana bayou works hard to make a life for his family, he must confront the ghosts of his past as he discovers that he could be deported from the only country he has ever called home.",
  "popularity": 41.969,
  "poster_path": "/a4xykTie8BOSW0y6K5u1AcmW4HW.jpg",
  "release_date": "2021-09-10",
  "title": "Blue Bayou",
  "video": false,
  "vote_average": 7.6,
  "vote_count": 130
}

it('renders without crashing', () => {
  const div = document.createElement("div");
  ReactDOM.render(<MovieItem movie={movie} />, div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<MovieItem movie={movie} />).toJSON();
  expect(tree).toMatchSnapshot();
});
