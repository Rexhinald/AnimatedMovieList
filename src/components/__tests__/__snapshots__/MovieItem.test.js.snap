// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="movie-item"
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "padding": 20,
    }
  }
>
  <img
    src="https://image.tmdb.org/t/p/original//a4xykTie8BOSW0y6K5u1AcmW4HW.jpg"
    style={
      Object {
        "width": 60,
      }
    }
  />
  <div
    className="movie-item-text"
  >
    <div
      className="movie-item-title"
    >
      Blue Bayou
    </div>
    <div
      style={
        Object {
          "transform": "none",
        }
      }
    >
      2021-09-10
    </div>
  </div>
</div>
`;
